barretenberg/cpp/src/barretenberg/commitment_schemes/shplonk/shplemini.hpp

```diff
@@ -439,5 +439,71 @@ template <typename Curve> class ShpleminiVerifier_ {
             commitments.emplace_back(std::move(fold_commitments[j]));
         }
     }
+
+    /**
+     * @brief Combines scalars of repeating commitments to reduce the number of scalar multiplications performed by the
+     * verifier.
+     *
+     * @details The Shplemini verifier gets the access to multiple groups of commitments, some of which are duplicated
+     * by design. This method combines the scalars associated with these repeating commitments, reducing the total
```

Comment:
will you please explain why here?

Reply:
expanded slightly
I think, the real answer - it's just easier to have those entities and shifts